### PR TITLE
[gcp] data/data: use gcs bucket for bootstrap ignition and expose bootstrap node

### PR DESCRIPTION
**Pull request by csrwng**
Moves bootstrap ignition to gcs bucket.

### CODE REVIEW

**Comment by csrwng** [comment]
@abhinavdahiya this would allow us to make progress even if it's not the final solution. It will make the ignition object public (otherwise the URL we use for ignition won't work)

File: data/data/gcp/bootstrap/main.tf [hunk 1]
```diff
@@ -1,3 +1,26 @@
...
+  name = "${var.cluster_id}-ignition"
+}
+
+resource "google_storage_bucket_object" "ignition" {
+  bucket  = "${google_storage_bucket.ignition.name}"
+  name    = "bootstrap.ign"
+  content = var.ignition
+}
+
+resource "google_storage_object_access_control" "ignition-public" {
```
Comment (abhinavdahiya) [inline data/data/gcp/bootstrap/main.tf]: this is not acceptable at all. We cannot make the bootstrap.ign contents public.

File: data/data/gcp/bootstrap/main.tf [hunk 2]
```diff
@@ -1,3 +1,26 @@
+resource "google_storage_bucket" "ignition" {
+  name = "${var.cluster_id}-ignition"
```
Comment (abhinavdahiya) [inline data/data/gcp/bootstrap/main.tf]: this should probably be `clsuterid-bootstrap-ignition`

File: data/data/gcp/bootstrap/main.tf [hunk 3]
```diff
@@ -1,3 +1,26 @@
...
+resource "google_storage_object_access_control" "ignition-public" {
+  object = "${google_storage_bucket_object.ignition.output_name}"
+  bucket = "${google_storage_bucket.ignition.name}"
+  role   = "READER"
+  entity = "allUsers"
+}
+
+data "ignition_config" "redirect" {
+  replace {
+    source = "http://storage.googleapis.com/${google_storage_bucket.ignition.name}/bootstrap.ign"
```
Comment (abhinavdahiya) [inline data/data/gcp/bootstrap/main.tf]: this should be https

File: data/data/gcp/bootstrap/main.tf (hunk 1 above)
Comment (csrwng) [reply data/data/gcp/bootstrap/main.tf]: agreed, but is this acceptable to unblock us while we get to a better solution?

**Comment by csrwng** [comment]
updated

**Comment by csrwng** [comment]
@abhinavdahiya updated to use a signed URL. However, for whatever reason I'm not able to fetch that URL locally. I always get `<?xml version='1.0' encoding='UTF-8'?><Error><Code>AccessDenied</Code><Message>Access denied.</Message><Details>Anonymous caller does not have storage.objects.get access to cewong-6l75q-bootstrap-ignition/bootstrap.ign.</Details></Error>`. Going to see if there's some parameter I'm missing in generating it.

**Comment by csrwng** [comment]
@abhinavdahiya looks like this is working fine. It must be an encoding thing when I copy it out of the metadata. Looking at the log of the vm when it's coming up, I see:
```
[    9.240767] ignition[674]: GET result: OK
[    9.242297] ignition[674]: GET https://storage.googleapis.com/cewong-qxfcx-bootstrap-ignition/bootstrap.ign?GoogleAccessId=[blahblah]: attempt #1
[    9.391515] ignition[674]: GET result: OK
```

**Pull request merged**